app/common/data/interfaces/collections.py

```diff
@@ -60,11 +61,27 @@ def update_collection_schema(schema: CollectionSchema, *, name: str) -> Collecti
     return schema
 
 
-def get_collection(collection_id: UUID) -> Collection:
-    return db.session.get_one(
-        Collection,
-        collection_id,
-    )
+def get_collection(collection_id: UUID, with_full_schema: bool = False) -> Collection:
+    options = []
+    if with_full_schema:
+        options.append(
+            joinedload(Collection.collection_schema)
+            .selectinload(CollectionSchema.sections)
```

Comment:
I'm sure you've tried a few options here but I think everything down to questions (maybe we'd want to stop at forms intentionally) here could be a single joined query 

I think there will be some changes around how sections are modeled and a few extra things to track as we factor in groups so no need to get this perfect from the get go - a good one to chat about when we get a minute though